### PR TITLE
Bringing map to pair coercion and basename from WDL4S

### PR DESCRIPTION
**Pull request by cjllanwarne**
Still needs
 - [x] Changelog

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala [hunk 1]
```diff
@@ -389,7 +389,8 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
         List.empty
       }
       WdlNamespaceWithWorkflow.load(w.wdlSource, importResolvers) match {
-        case Failure(e) => s"Unable to load namespace from workflow: ${e.getMessage}".invalidNel
+        case Failure(e) =>
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala]: Revert this

File: supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala [hunk 2]
```diff
@@ -191,7 +191,7 @@ class JesAsyncBackendJobExecutionActor(override val standardParams: StandardAsyn
   }
 
   private[jes] def generateJesOutputs(jobDescriptor: BackendJobDescriptor): Set[JesFileOutput] = {
-    val wdlFileOutputs = call.task.findOutputFiles(jobDescriptor.fullyQualifiedInputs, NoFunctions) map relativeLocalizationPath
+    val wdlFileOutputs = call.task.findOutputFiles(jobDescriptor.fullyQualifiedInputs, PureStandardLibraryFunctions) map relativeLocalizationPath
```
Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: Could you explain why this change? 

(personal interest/learning purposes, not a critique)

Comment (cjllanwarne) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: It's because otherwise we weren't able to get engine function evaluations showing up as file outputs. 
E.g. in my motivating situation
```
output {
  String x = read_string(basename(foo, ".bam") + ".bai")
}
```
With no functions, that'll fail to evaluate and the file won't be delocalized for us
With pure functions, things that don't interact externally can still be evaluated, but not the full suite of functions (e.g. we still can't do `read_string(read_string(foo))`)

Comment (geoffjentry) [inline supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesAsyncBackendJobExecutionActor.scala]: Gotcha, thanks

**Comment by Horneth** [comment]
👍

**Comment by geoffjentry** [comment]
👍
